### DISCUSSION TITLE
Creating video elements without specifying the source file

### DISCUSSION THREAD

**Discussion by ercanhocaninyeri** [zauberzeug/nicegui#1948] [Ideas / Feature Requests]
I would like to open a video file in the GUI after selecting the video using the [file picker](https://github.com/zauberzeug/nicegui/tree/main/examples/local_file_picker). The video is to be displayed in a [layout element ](https://nicegui.io/documentation#column_element)and every time user selects another video, it will be played in the same dedicated area of the GUI. Therefore I need to define a video element without explicitly stating the source file and set the value for the source file according to the user's selection. 
I mean something like this:
```py
# create the video element:
v = ui.video(autoplay=True).style('position:absolute; left:350px; top:120px; height:270px; width:480px;') 
# specify the video file:
v.src = path_to_video
```

**Comment by falkoschindler** [comment]
Yes, it would be useful to be able to change the video source later. Maybe `ui.video` should derive from `SourceElement`?

Here is a potential workaround until this feature is implemented:

```py
v = ui.video('')
ui.button('Set source',
          on_click=lambda: v.props('src=https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/360/Big_Buck_Bunny_360_10s_1MB.mp4'))
```
